File: swbasic/mathematics/1978/1978.py
import math

def isPrime(a):
    i = 2
    iPrime = True
    if a == 1:
        iPrime = False
        return iPrime

    while i<= int(math.sqrt(a)):
        if a % i == 0:
           iPrime = False
           return iPrime
        i += 1
    return iPrime

if __name__ == "__main__":
    n = int(input())
    a = []
    a = list(input().split())
    count = 0
    for i in range(n):
        if isPrime(int(a[i])):
            count += 1
    print(count)

File: swbasic/mathematics/1978/1978.c
#include <stdio.h>

int isPrime(int a)
{
    int i = 2;
    int iPrime = 1;

    if (a == 1)
    {
        iPrime = 0;
        return iPrime;
    }

    while (i < a) // sqrt 사용시 처리 속도 향상 기대 가능
    {
        if (a % i == 0)
        {
            iPrime = 0;
            break;
        }
        i += 1;
    }
    return iPrime;
}

int main()
{
    int n;
    scanf("%d", &n);
    int a[n];

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &a[i]);
    }

    int count = 0;

    for (int j = 0; j < n; j++)
    {
        if (isPrime(a[j]))
        {
            count++;
        }
    }
    printf("%d\n", count);

    return 0;
}